File: javascript/medium/jumpGame.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let j = nums.length-1, i = j-1;
    while(i >= 0) {
        if((i + nums[i]) >= j) {
            j = i;
        }
        i--;
    }
    return j === 0;
};

File: ruby/medium/jump_game.rb
# @param {Integer[]} nums
# @return {Boolean}
def can_jump(nums)
    i, j = nums.length-2, nums.length-1
    while i >= 0
        j = i if((nums[i] + i) >= j)
        i -= 1
    end
    return j == 0
end
